### PR TITLE
Feature/32 private profile header

### LINKED ISSUE #32: Private profile header
Image
Image

https://www.figma.com/file/DTJwC2mQ9kvyjM0171tqHR/airswap-web?node-id=77%3A1014&t=vRYyxxtVHOy46n59-0

### PR DESCRIPTION
**Pull request by GiuntaLucas**
# Description

Your PR should at least have a ticket number linked to it.

Example:

Fixes #32

A short video or an image demonstrating the implementation is also helpful. Especially for designers checking your PR.

Create a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) if you're looking for feedback but not ready for a final review.

# Steps to test the change

Help the reviewer test what you implemented/changed. Include all the steps needed to perform the test.

Example:

1. Add environment variable `REACT_APP_VAR_X=value` in .env file
2. Run `yarn start`
3. Go to home page
4. Click on an NFT
5. Check component XYZ

# Checklist:

- [ ] I have read and understood [CONTRIBUTING.MD](https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md)
- [ ] I have included the ticket number in my PR
- [ ] I have performed a self-review of my own code
- [ ] I have commented the parts where my code is hard to understand

### CODE REVIEW

File: src/widgets/ProfileWidget/ProfileWidget.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            line-height: 1.5rem;
+            color: var(--c-light-grey);
+            background: var(--c-dark);
+
+            &--active {
+                background-color: var(--c-grey);
+                color: #fff;
+            }
+        }
+    }
```
Comment (piersss) [inline src/widgets/ProfileWidget/ProfileWidget.scss:41]:
```suggestion
&__button-group {
        display: flex;
        border: solid 1px var(--c-grey);
        border-radius: 4px;

        @include for-size(desktop-up) {
            width: 19.5rem;
            transform: translateY(-2.56rem);
        }
    }

    &__button-group__button {
        width: -webkit-fill-available;
        height: 2rem;
        font-weight: 700;
        font-size: .75rem;
        line-height: 1.5rem;
        color: var(--c-light-grey);
        background: var(--c-dark);

        &--is-active {
            background-color: var(--c-grey);
            color: #fff;
        }
    }
```

https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md#naming-of-modifiers

File: src/widgets/ProfileWidget/ProfileWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      // must remove the hardcoded value before to raise the pr
+        backgroundImage={collectionImage || 'collection/collection-example-image.png'}
+        avatarUrl={avatarUrl}
+        ensAddress={ensAddress}
+        address={account || ''}
+        onLogoutButtonClick={handleDisconnectClick}
+      />
+      <div className="profile-widget__button-group-container">
+        <div className="profile-widget__button-group">
+          <Button text="NFTs" className="profile-widget__button-group__button profile-widget__button-group__button--active" />
```
Comment (piersss) [inline src/widgets/ProfileWidget/ProfileWidget.tsx:35]:
```suggestion
          <Button text="NFTs" className="profile-widget__button-group__button profile-widget__button-group__button--is-active" />
```

File: src/widgets/ProfileWidget/subcomponents/ProfileHeader/ProfileHeader.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  height: 9rem;
+  width: 100%;
+  // position: absolute;
+  display: flex;
+  flex-direction: column;
+  justify-content: end;
+  // left: 0;
+  // top: 0;
+  background-size: cover;
+  background-repeat: no-repeat;
```
Comment (piersss) [inline src/widgets/ProfileWidget/subcomponents/ProfileHeader/ProfileHeader.scss:13]:
```suggestion
  display: flex;
  flex-direction: column;
  justify-content: end;
  height: 9rem;
  width: 100%;
  background-size: cover;
  background-repeat: no-repeat;
```

https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md#order-of-properties

File: src/widgets/ProfileWidget/ProfileWidget.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
+@import "src/styles/index";
+
+.profile-widget {
+    display: grid;
```
Comment (piersss) [inline src/widgets/ProfileWidget/ProfileWidget.scss:4]: Don't use display grid for things that are not grids.

File: src/widgets/ProfileWidget/ProfileWidget.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        @include for-size(desktop-up) {
+            display: flex;
+            justify-content: center;
+        }
+    }
+
+    &__button-group {
+        border: solid 1px var(--c-grey);
+        border-radius: 4px;
+        display: flex;
```
Comment (piersss) [inline src/widgets/ProfileWidget/ProfileWidget.scss:20]:
```suggestion
        display: flex;
        border: solid 1px var(--c-grey);
        border-radius: .25rem;
```

File: src/widgets/ProfileWidget/ProfileWidget.scss [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (piersss) [inline src/widgets/ProfileWidget/ProfileWidget.scss:21]:
```suggestion
@include for-size(tablet-portrait-up) {
    height: 17.5rem;
  }

```

**Review by piersss — CHANGES_REQUESTED** [review]
Padding on mobile not correct. Look at CollectionPage how it's done.

Scss tab spacing should be 4.

**Comment by GiuntaLucas** [comment]
@piersss  Thank's for your comments, i will have a look deeply tonight or tomorrow. Already push your css fix.

File: src/widgets/ProfileWidget/subcomponents/ProfileHeader/ProfileHeader.scss [hunk 7]
```diff
@@ -0,0 +1,21 @@
+@import "src/styles/index";
+
+.profile-header {
+  display: flex;
+  flex-direction: column;
+  justify-content: end;
```
Comment (piersss) [inline src/widgets/ProfileWidget/subcomponents/ProfileHeader/ProfileHeader.scss:6]:
```suggestion
  justify-content: flex-end;
```

File: src/widgets/ProfileWidget/ProfileWidget.scss [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        line-height: 1.5rem;
+        color: var(--c-light-grey);
+        background: var(--c-dark);
+
+        &--is-active {
+            background-color: var(--c-grey);
+            color: #fff;
+        }
+    }
+}
```
Comment (Goddak) [inline src/widgets/ProfileWidget/ProfileWidget.scss:44]:
```suggestion
}

```

Newline at end of file

File: src/widgets/ProfileWidget/subcomponents/ProfileHeader/ProfileHeader.scss [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	@include for-size(desktop-up) {
+		height: 27.5rem;
+		border-radius: .5rem;
+		box-shadow: 0px 4px 45px -32px rgba(0, 0, 0, 0.99);
+	}
+
+	&__wallet-info {
+		padding: 1.625rem;
+	}
+}
```
Comment (Goddak) [inline src/widgets/ProfileWidget/subcomponents/ProfileHeader/ProfileHeader.scss:27]:
```suggestion
}

```

New line at end of file

File: src/components/WalletInfo/WalletInfo.scss [hunk 10]
```diff
@@ -12,7 +12,7 @@
   }
 
   &--is-banner {
```
Comment (Goddak) [inline src/components/WalletInfo/WalletInfo.scss:14]: We no longer need this class, all it was doing was setting a bgColor so if you don't need/want it anymore then remove the class.

File: src/widgets/ProfileWidget/ProfileWidget.scss [hunk 11]
```diff
@@ -0,0 +1,44 @@
+@import "src/styles/index";
+
+.profile-widget {
+
+    &__button-group-container {
+        display: block;
```
Comment (Goddak) [inline src/widgets/ProfileWidget/ProfileWidget.scss:6]: This is default behaviour you don't need to specify

**Review by Goddak — CHANGES_REQUESTED** [review]
Before you continue coding; take the time to install & configure ESLint. There is a config provided within the project that should be automatically used and will save a lot of trivial comments when reviewing your work.

Next up, before you continue; take a look at how the other widgets are sized using `calc()`, you shouldn't need to be positioning things with large 9rem margins and absolute positioning for a background image (just remove the padding from the parent div and apply it to it's child).

**Comment by GiuntaLucas** [comment]
@Goddak I will have a look to eslint... for the calc, I k now it, but as the image height is fix, I dont have to use Calc. For the padding, the padding is set into the main component Page which is the same for every page. Plus, except for the image, the padding still need to be apply for the rest of the element.. For the description of the pr, we'll, it is on the issue linked.

**Comment by Goddak** [comment]
> @Goddak I will have a look to eslint... 

Nice, it's essential for working on the project really, consistent code is important.

>for the calc, I k now it, but as the image height is fix, I dont have to use Calc. For the padding, the padding is set into the main component Page which is the same for every page. Plus, except for the image, the padding still need to be apply for the rest of the element, we dont have to change the behavior of a parent component who are using transclusion.

Yes it is, so go and look at the collection page which has a similar setup to yours and copy that rather than what you have done.

>  For the description of the pr, we'll, it is on the issue linked.

Yes but again, look around at what others are doing and join in, a short description and maybe an image or video like Piers has been championing lately.. BUT most importantly is the `fixes` text I have now added. It takes a few moments to improve the life of those you work with.

File: src/components/WalletInfo/WalletInfo.scss (hunk 10 above)
Comment (GiuntaLucas) [reply src/components/WalletInfo/WalletInfo.scss:14]: i guess this change will be require when the pr will be ready to merge (resolve conflict with develop branch).
For now, i still have the isBanner prop from the WalletInfo component, will fit it once the merge with develop done.

Comment (piersss) [reply src/components/WalletInfo/WalletInfo.scss:14]: @Goddak Yeah not a fan, can you change it maybe in your branch?

**Review by piersss — APPROVED** [review]
LGTM

**Pull request merged**